app/src/main/java/io/gnosis/safe/ui/signing/owners/OwnerPagingProvider.kt

```diff
@@ -0,0 +1,34 @@
+package io.gnosis.safe.ui.signing.owners
+
+import androidx.paging.Pager
+import androidx.paging.PagingConfig
+import androidx.paging.PagingData
+import io.gnosis.safe.utils.MnemonicAddressDerivator
+import kotlinx.coroutines.flow.Flow
+import pm.gnosis.model.Solidity
+
+class OwnerPagingProvider(
+    private val derivator: MnemonicAddressDerivator
+) {
+
+    fun getOwnersStream(): Flow<PagingData<Solidity.Address>> {
+        return Pager(
+            initialKey = 0,
+            config = PagingConfig(
+                pageSize = PAGE_SIZE,
+                prefetchDistance = 1,
+                enablePlaceholders = false,
+                initialLoadSize = PAGE_SIZE,
+                maxSize = PAGE_SIZE * PAGES_THRESHOLD
+            ),
+            pagingSourceFactory = {
+                OwnerPagingSource(derivator, PAGES_THRESHOLD)
```

Comment:
I don't understand the word threshold here. The parameter is called maxPages. That makes more sense.